packages/vite/src/node/server/index.ts

```diff
@@ -957,7 +969,8 @@ export async function restartServerWithUrls(
     server: { port, host },
   } = server.config
   if (
-    (port ?? DEFAULT_DEV_PORT) !== (prevPort ?? DEFAULT_DEV_PORT) ||
+    (_currentServerPort ?? port ?? DEFAULT_DEV_PORT) !==
```

Comment:
Actually I think this not needed and this PR fixed a case where the URL could change without a change in the configuration (that would not trigger a re-print or urls)

Reply:
I was checking the same. If it is needed, I think we should review how we are overwriting `config.server.port` here:
https://github.com/vitejs/vite/blob/e5094e5bf2aee3516d04ce35ba2fb27e70ea9858/packages/vite/src/node/server/index.ts#L652

Once `await server.restart()` resolves, `config.server.port` should already have the updated port value or this API would be quite hard to use externally.

Ok to do this in another PR if this isn't working already though.

Reply:
I'm not sure about overriding the config, I think it's good to know what the user set and what is actually used. we could make this property available on the server object I think

Reply:
We have always overrided the config, so stop doing it would be quite a breaking change. I'm only referring to the timing of when this override is done.

Reply:
Oh ok yeah my bad so it _should_ be cleaner to move some of the logic there

Reply:
After I merged the code on the main branch, I found that the port had been updated as @patak-dev  said. 
But now it seems that a new problem has arisen, `server.resolvedUrls` has become null, and the URL information will still be printed out repeatedly.

Reply:
This seems to be unrelated to this PR, no? Probably a regression in Vite 5.

Reply:
@btea this should fix the `resolvedUrls` regression (and I think the `port` should be correct too)
- https://github.com/vitejs/vite/pull/14890

So I think you should be able to remove these changes after it is merged.

Reply:
I've removed those changes and now the port is correct as you said and the other work you have fixed in the new pr. 👍